### ISSUE TITLE
Attention masks for fine-tuning

### ISSUE THREAD

**Issue by andreasgrv** [OpenEvaByte/evabyte#6]
Hi,

Thanks for the releasing the models and code, cool project and a lot of food for thought!

### Attention masks for fine-tuning
I am currently trying to fine-tune a different head on top of EvaByte for SFT, and I am running into issues with the masking tensor when I try to use an attention_mask from a huggingface dataloader that is not None. For the training branch of the code [you expect attention_mask to be a tuple](https://github.com/OpenEvaByte/evabyte/blob/98d5f48d32197b803e7560a798da35c7a4bdcf4d/evabyte_hf/eva.py#L149)  of three masking tensors, `(window_mask, chunk_mask, intra_chunk_mask)`.

> Is there a function I can call to produce these given the huggingface mask tensor?

The closest I could find was [_prepare_eva_generation_attn_mask_triton](https://github.com/OpenEvaByte/evabyte/blob/98d5f48d32197b803e7560a798da35c7a4bdcf4d/evabyte_hf/modeling_evabyte.py#L416), but this function returns a tuple of two attention masks if the model is in training mode.

> Could you maybe elaborate briefly on the differences between the window mask, the chunk mask and the intra chunk mask?

Thank you!

**Comment by LZhengisme** [comment]
Hey Andreas,

Thanks and glad you're enjoying the project! It seems the utility script for creating training attention masks got accidentally deleted during refactoring :( I've just pushed a self-contained script here (https://github.com/OpenEvaByte/evabyte/blob/main/training_utils.py) with detailed docs inside, which should help.

Quick context on the tuple-based attention masks: these are mainly for **packed fine-tuning/pretraining**, ensuring that tokens from different documents don't attend across their respective boundaries. If you're not doing packed fine-tuning, you can simply pass attention_mask=None, which should be much simpler.

BTW, EvaByte is extensively tested with input sequences whose lengths are multiples of window_size (i.e., 2048). Other lengths are also supported but have much less thorough testing—so to stay safe, it's best to **right-pad sequences to multiples of 2048**. This won't increase computational cost significantly and should avoid potential issues.

Hope this helps and let me know if you run into any other questions!
